### PR TITLE
Remove kube-rbac-proxy

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
kube-rbac-proxy sidecar container is not longer needed as metrics auth is handled within controller runtime. It has been removed in cluster-api and following the same.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Remove kube-rbac-proxy
```

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -28,7 +28,8 @@ spec:
         args:
         - "--leader-elect"
         - "--provider-id-fmt=${PROVIDER_ID_FORMAT:=v2}"
-        - "--metrics-bind-addr=127.0.0.1:8080"
+        - "--diagnostics-address=${CAPA_DIAGNOSTICS_ADDRESS:=:8443}"
+        - "--insecure-diagnostics=${CAPA_INSECURE_DIAGNOSTICS:=false}"
```
Comment (mkumatag) [inline config/manager/manager.yaml:32]: why `CAPA_DIAGNOSTICS_ADDRESS`? why not `CAPIBM_DIAGNOSTICS_ADDRESS`

 `CAPA_INSECURE_DIAGNOSTICS` => `CAPIBM_INSECURE_DIAGNOSTICS `???

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -37,6 +38,10 @@ spec:
           - containerPort: 9440
             name: healthz
             protocol: TCP
+          - containerPort: 8443
+            name: metrics
+            protocol: TCP
+        readinessProbe:
```
Comment (mkumatag) [inline config/manager/manager.yaml:44]: why this redundant `readinessProbe`?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/manager/manager.yaml (hunk 2 above)
Comment (Amulyam24) [reply config/manager/manager.yaml:44]: typo, fixed it.

File: config/manager/manager.yaml (hunk 1 above)
Comment (Amulyam24) [reply config/manager/manager.yaml]: Fixed it.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**